pkg/testing/tools/elasticsearch.go

```diff
@@ -103,11 +173,158 @@ func GetIndicesWithContext(ctx context.Context, client elastictransport.Interfac
 	return respData, nil
 }
 
+// CreateAPIKey creates an API key with the given request data
+func CreateAPIKey(ctx context.Context, client elastictransport.Interface, req APIKeyRequest) (APIKeyResponse, error) {
+	var buf bytes.Buffer
+	err := json.NewEncoder(&buf).Encode(req)
+	if err != nil {
+		return APIKeyResponse{}, fmt.Errorf("error creating ES query: %w", err)
+	}
+
+	apiReq := esapi.SecurityCreateAPIKeyRequest{Body: &buf}
+	resp, err := apiReq.Do(ctx, client)
+	if err != nil {
+		return APIKeyResponse{}, fmt.Errorf("error creating API key: %w", err)
+	}
+	resultBuf, err := handleResponseRaw(resp)
+	if err != nil {
+		return APIKeyResponse{}, fmt.Errorf("error handling HTTP response: %w", err)
+	}
```

Comment:
`handleResponseRaw` does not close the response body
```suggestion
	defer resp.Body.Close()
	resultBuf, err := handleResponseRaw(resp)
	if err != nil {
		return APIKeyResponse{}, fmt.Errorf("error handling HTTP response: %w", err)
	}
```